packages/logger/lib/src/log_level.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_logger/src/ansi_constants.dart';
+import 'package:ansicolor/ansicolor.dart';
+
+/// An enumeration of the different levels of logging.
+/// The levels are progressive, with lower-value items being lower priority
+enum LogLevel implements Comparable<LogLevel> {
+  /// Logs for showing behavior of particular components/flows
+  /// Not suitable for production as they may contain sensitive information
+  verbose,
+
+  /// Logs for understanding system behavior
+  /// May contain information inappropriate for emission into production environments
+  debug,
+
+  /// Logs providing terse info about general operation and flow of software
+  info,
+
+  /// Logs indicating potential issues
+  warn,
+
+  /// Logs when system is not operating as expected
+  error,
+
+  /// Highest priority log to use as threshold value to prevent any logs from being emitted
+  none;
+
+  int compareTo(LogLevel other) {
+    return this.index - other.index;
+  }
+
+  bool operator >(LogLevel value) => index > value.index;
+  bool operator >=(LogLevel value) => index >= value.index;
+  bool operator <(LogLevel value) => index < value.index;
+  bool operator <=(LogLevel value) => index <= value.index;
+
+  /// Creates a formatted string based on the log level.
+  String toFormattedString() {
```

Comment:
Jordan - I missed this in the first review. I think the default formatting can be simplified a bit. We want to use color and formatting to draw attention to things that need it. If we use it everywhere, the formatting loses meaning. Personally I would leave anything other than warn/error as unformatted. Warn/error obviously should have some attention drawn to it. Red/yellow makes sense, but personally I think the background colors are a bit much.

Reply:
@dnys1 what are your thoughts on Jordan's comments? I understand you've created a lot of Flutter applications, so you probably have the most experience dealing with logs from multiple libraries and your own code ...

To mitigate confusion, the less important log messages have a grey or blue background. The reason behind this was to differentiate default logs from Amplify from other logs that might be coming from other libraries / packages / native code that we or the customer might encounter when running the program.